src/bcf-toolkit/Converter/Bcf21/Converter.cs

```diff
@@ -63,150 +73,43 @@ public class Converter : IConverter {
     };
 
     // Writing bcf files
-    await WriteBcf(bcf, target);
+    await FileWriter.WriteBcf(bcf, target);
   }
 
-  public async Task<Stream> ToBcfStream(IBcf bcf) {
+  public async Task<Stream> ToBcfStream(IBcf bcf, BcfVersionEnum targetVersion) {
+    // Convert the bcf to the target version
+    var converterFn = _converterFnMapper[targetVersion];
+    var convertedBcf = converterFn((Bcf)bcf);
+
+    // Write the converted bcf to tmp and create file stream
     var workingDir = Directory.GetCurrentDirectory();
     var bcfTargetPath = workingDir + "/bcf.bcfzip";
-
-    var tmpFolder = await WriteBcf((Bcf)bcf, bcfTargetPath, false);
-
+    var writerFn = _writerFnMapper[targetVersion];
+    var tmpFolder = await writerFn(convertedBcf, bcfTargetPath, false);
     var stream = new FileStream(bcfTargetPath, FileMode.Open, FileAccess.Read);
 
-    // After the filestream is ready we can delete the folders
+    // After the stream is ready the folders should be deleted
```

Comment:
A general feedback on all the comments here, I think some are quite verbose. I think a comment should rather say why the code does something and not what the code does. Because the code itself should say already what it does (if it's not super cryptic or an algorithm, or something like that).

Reply:
True, will be extended!